util/src/main/java/io/kubernetes/client/informer/cache/Controller.java

```diff
@@ -65,19 +66,34 @@
 
   private ScheduledFuture reflectorFuture;
 
+  /* visible for testing */ BiConsumer<Class<ApiType>, Throwable> exceptionHandler;
+
   public Controller(
```

Comment:
I am not sure if these constructor chains are a good strategy long term. The best option I currently see is
- add a builder for the class
- use the all params constructor from the builder
- deprecate all constructors with partial param sets

Also I would try to get rid of the many new calls here via factories. This could improve testability and introduce entry points for people who want to influence behaviour. A larger change but could be done in baby steps. What do you think @brendandburns ?

Reply:
Can you file this as an issue so we can have a broader discussion? I want to get @yue9944882 to weigh in as well.

Reply:
Yay, have created issue #1995 for discussion. Looking forward to if :)